src/rgw/rgw_rest_pubsub.cc

```diff
@@ -1215,21 +1417,34 @@ void RGWPSListNotifsOp::execute(optional_yield y) {
     return;
   }
 
-  const RGWPubSub ps(driver, s->owner.id.tenant);
-  const RGWPubSub::Bucket b(ps, bucket.get());
-  
   // get all topics on a bucket
   rgw_pubsub_bucket_topics bucket_topics;
-  op_ret = b.get_topics(this, bucket_topics, y);
-  if (op_ret < 0) {
-    ldpp_dout(this, 1) << "failed to get list of topics from bucket '" << s->bucket_name << "', ret=" << op_ret << dendl;
-    return;
+  if (s->penv.site->get_zonegroup().supports(
+          rgw::zone_features::notification_v2)) {
+    rgw::sal::Attrs attrs = bucket->get_attrs();
+    auto iter = attrs.find(RGW_ATTR_BUCKET_NOTIFICATION);
+    if (iter == attrs.end()) {
+      // no notification found
+      return;
+    }
+    bufferlist& bl = iter->second;
+    auto biter = bl.cbegin();
+    bucket_topics.decode(biter);
```

Comment:
i thought about it, but since this is reading from the rados object, so if this was not stored in correct way it is a problem in first place
so i have a check while storing ane encoding.. if that passes then we can safely assume retrieving and decoding should not have a problem>

Reply:
there could be a case where changed format between versions but did that in non-backward-compatible way.
even though should be considered a "software bug", we don't want the RGW to continously crash (due to unhandled exception) in this case.
so, IMO, we should `try/catch` around this `decode` call

Reply:
right, a lot can change in the time between the encode and the decode. all decodes are expected to handle buffer::error exceptions

Reply:
> right, a lot can change in the time between the encode and the decode. all decodes are expected to handle buffer::error exceptions

Done